warehouse/query-core/src/main/java/datawave/query/iterator/SourceManager.java

```diff
@@ -105,40 +132,48 @@ public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> op
     @Override
     public void next() throws IOException {
         if (null != child) {
+            if (lastRange == null) {
+                throw new IllegalStateException("seek must be called prior to next()");
+            }
             
             reseekIfNeeded();
             
             child.next();
             
             try {
                 lastKey = child.getTopKey();
+                lastValue = child.getTopValue();
+                
+                // need to understand when to expect no more keys for this range to differentiate from the start case
+                if (lastKey == null) {
+                    rangeDone = true;
```

Comment:
set lastValue to null here or move the child.getTopValue() into and else clause

Reply:
lastValue should be set to null already if lastKey is set to null